app/controllers/sessions/edit_controller.rb

```diff
@@ -104,20 +104,17 @@ def set_patients
     # Get a list of patients but ensure we don't include patients that are
     # already in other sessions, if those sessions are active (not draft) and
     # for the same vaccine/programme.
+    patients_in_active_sessions_and_same_programme =
+      Patient
+        .joins(:patient_sessions)
+        .joins(:sessions)
+        .where(sessions: { active: true, programme: @session.programme })
     @patients =
       @session
         .location
         .patients
-        .where.not(
-          Session
-            .joins(:patient_sessions)
-            .active
-            .where(programme: @session.programme)
-            .where("patient_sessions.patient_id = patients.id")
-            .arel
-            .exists
-        )
-        .sort_by(&:last_name)
+        .where.not(id: patients_in_active_sessions_and_same_programme)
+        .order(:last_name)
```

Comment:
@thomasleese I haven't tested this, I probably should, but is it possible that the reason we had `sort_by` is because `order` won't work due to encryption?

Reply:
Hmm, yeah, that's very possible, I'm not actually sure myself why we would've used `sort_by` here and not `order`.

Reply:
Confirmed that `order` does not work.